### PR TITLE
Fix cloning regions

### PR DESCRIPTION
**Pull request by timoludwig**
### Short description

I noticed that #1259 also affected the language tree

### Proposed changes

- Improve logging representation of model objects
- Add tests for cloning regions
    - Add test case which fails in the current state to reproduce #1259
    - Order translation models by language to keep deterministic order to compare querysets
- Fix duplication of language tree
    - Commit https://github.com/digitalfabrik/integreat-cms/commit/f708ba12b2fa49163cc284ba0fd250e7f9736219 only handled the page tree, not the language tree
    - Make test case added in the previous commit succeed
    - Add unnecessarily beautiful logging output

### Resolved issues

Fixes: #1259 (again)

### CODE REVIEW

**Comment by svenseeberg** [comment]
The following SQL query will fix broken databases:
```sql
UPDATE cms_languagetreenode SET tree_id=(SELECT tree_id+1 FROM cms_languagetreenode ORDER BY tree_id DESC LIMIT 1) WHERE id IN (SELECT id FROM (SELECT MAX(id) AS id, tree_id, MAX(region_id) AS region_id, COUNT(*) as duplicate_count, lft FROM cms_languagetreenode GROUP BY lft,tree_id ORDER BY lft DESC) AS p WHERE duplicate_count>1);
```

**Comment by svenseeberg** [comment]
@timoludwig would it make sense to add a unique constraint for the tuple `(tree_id, lft)` ?

**Review by svenseeberg — CHANGES_REQUESTED** [review]
Just one very minor suggestion.

File: integreat_cms/cms/models/users/user.py [hunk 1]
```diff
@@ -96,10 +98,27 @@ def role(self):
...
+        # Many-to-many relationships can only be used for objects that are already saved to the database
+        if self.id:
+            groups = self.groups.all()
+            if groups:
+                # Assume users only have one group/role
+                return groups[0].role
+        return None
+
+    @cached_property
+    def primary_region(self):
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply integreat_cms/cms/models/users/user.py]: Yeah, makes sense :+1:

**Comment by timoludwig** [comment]
> would it make sense to add a unique constraint for the tuple `(tree_id, lft)` for all trees ?

@svenseeberg Good idea! However, this doesn't seem to work since the tree library violates these constraints temporarily during move operations for performance reasons. I did not find a way to ignore these constraints during moving, probably this isn't possible because they are enforced on the database layer. It seems that we have to keep an eye on our database structure and fix problems as they arise :see_no_evil:

**Review by svenseeberg — APPROVED** [review]

**Pull request merged**